multicluster/service-mirror/cluster_watcher.go

```diff
@@ -939,7 +939,7 @@ func (rcsw *RemoteClusterServiceWatcher) repairEndpoints(ctx context.Context) er
 		rcsw.log.Errorf("Failed to list mirror services: %s", err)
 	}
 	for _, svc := range mirrorServices.Items {
-		updatedService := svc
+		svc := svc
```

Comment:
Do we actually need to "pin" this? It seems that in Go, whenever we use an index and a value in a for loop, the element is automatically copied. So `svc` in this case can be passed along and operated on inside the loop without affecting the element in the original array (`mirrorServices.Items`).

e.g:

[internet example](https://go.dev/play/p/4b181zkB1O)
[trivial example created by me](https://go.dev/play/p/Ub6ihTiEY-3)

Think we might be able to just remove this completely? Not sure if redundant or not, wdyt?

Reply:
Yep so we do need to pin this. In a for loop, the `svc` variable from `for _, svc :=` is a single address that gets its value changed in each iteration. So while the value of `svc` changes, its address is always the same.

We would be okay without pinning if we didn't pass it by reference in `Ports:     rcsw.getEndpointsPorts(&svc)`. Without the pin, a lint warning is issued explaining that the use of `&svc` is essentially the same for each iteration and therefore it's implicit memory aliasing.